### DISCUSSION TITLE
[Question] In Navigation and Routing, how to use AlertDialog?

### DISCUSSION THREAD

**Discussion by jjjunehkim** [flet-dev/flet#3077] [Q&A] [answered]
### Question

I want to open ft.AlertDialog(dlg) when I click <ElevatedButton("Open Dlg")> in /store.
when I click <ElevatedButton("Open Dlg")>in /store, it  it doesn't work.
AlertDialog just opened after click <ElevatedButton("Open Dlg")> and click <ElevatedButton("Go Home")>

please help me!!!

### Code sample

```python
import flet as ft
import time

def main(page: ft.Page):
    page.title = "Navigation and Routing and AlertDialog examples"

    def open_dlg(e):
        page.dialog = dlg
        dlg.open = True
        page.update
        time.sleep(5)
        dlg.open = False
        page.update()

    dlg = ft.AlertDialog(
        title=ft.Text("Hello, you!"), on_dismiss=lambda e: print("Dialog dismissed!"),
    )

    def route_change(route):

        page.views.clear()
        page.views.append(
            ft.View(
                "/",
                [
                    ft.AppBar(title=ft.Text("Flet app"), bgcolor=ft.colors.SURFACE_VARIANT),
                    ft.ElevatedButton("Visit Store", on_click=lambda _: page.go("/store")),
                    # ft.ElevatedButton("Open Dlg", on_click=open_dlg),
                ],
            )
        )
        if page.route == "/store":

            page.views.append(
                ft.View(
                    "/store",
                    [
                        ft.AppBar(title=ft.Text("Store"), bgcolor=ft.colors.SURFACE_VARIANT),
                        ft.ElevatedButton("Open Dlg", on_click=open_dlg),
                        ft.ElevatedButton("Go Home", on_click=lambda _: page.go("/")),
                    ],
                )
            )
        page.update()

    def view_pop(view):
        page.views.pop()
        top_view = page.views[-1]
        page.go(top_view.route)

    page.on_route_change = route_change
    page.on_view_pop = view_pop
    page.go(page.route)


ft.app(target=main, host = "192.168.35.35", port = 8550, view=ft.AppView.WEB_BROWSER)
# ft.app(target=main, host = "192.168.35.35", port = 8550, view=ft.WEB_BROWSER)
```

### ------------------------------------------------------

- [x] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by abdokhalil07** [comment]
import flet as ft
import time

def main(page: ft.Page):
    page.title = "Navigation and Routing and AlertDialog examples"

    def open_dlg(e):
        page.dialog = dlg
        dlg.open = True
        page.update()
        close_dlg()
        
    def close_dlg():
        time.sleep(3)
        dlg.open = False
        page.update()

    dlg = ft.AlertDialog(
        title=ft.Text("Hello, you!"), on_dismiss=lambda e: print("Dialog dismissed!"),
    )

    def route_change(route):

        page.views.clear()
        page.views.append(
            ft.View(
                "/",
                [
                    ft.AppBar(title=ft.Text("Flet app"), bgcolor=ft.colors.SURFACE_VARIANT),
                    ft.ElevatedButton("Visit Store", on_click=lambda _: page.go("/store")),
                    # ft.ElevatedButton("Open Dlg", on_click=open_dlg),
                ],
            )
        )
        if page.route == "/store":

            page.views.append(
                ft.View(
                    "/store",
                    [
                        ft.AppBar(title=ft.Text("Store"), bgcolor=ft.colors.SURFACE_VARIANT),
                        ft.ElevatedButton("Open Dlg", on_click=open_dlg),
                        ft.ElevatedButton("Go Home", on_click=lambda _: page.go("/")),
                    ],
                )
            )
        page.update()

    def view_pop(view):
        page.views.pop()
        top_view = page.views[-1]
        page.go(top_view.route)

    page.on_route_change = route_change
    page.on_view_pop = view_pop
    page.go(page.route)


ft.app(target=main,  port = 8550, view=ft.AppView.WEB_BROWSER)
# ft.app(target=main, host = "192.168.35.35", port = 8550, view=ft.WEB_BROWSER)

**Reply by abdokhalil07** [reply, marked as answer]
dont make open and close dialog in the same function
write to open dialog  function and close dialog function
and call close_dialog from open_dialog

**Reply by jjjunehkim** [reply]
Thank you for your answering!!
Today! I found my code was wrong!!

I didn't insert parentheses as -> page.update()
I did  --> page.update

    def open_dlg(e):
        page.dialog = dlg
        dlg.open = True
        **page.update**  -> my mistake
        time.sleep(5)
        dlg.open = False
        page.update()

Thank you so much. I catched my mistake thanks to your reply
